Porting/sync-with-cpan

```diff
@@ -538,23 +600,38 @@ if ($$info {CUSTOMIZED}) {
 }
 
 chdir "..";
-if (@commit || @gone) {
-    say "Fixing MANIFEST";
+{
+    # we update the MANIFEST file always now, so that we can
+    # ensure each file from this sync is updated to say that we
+    # got it from the latest version.
+    say "Updating the MANIFEST file";
     my $MANIFEST     = "MANIFEST";
     my $MANIFEST_NEW = "$MANIFEST.new";
 
     open my $orig, "<", $MANIFEST
         or die "Failed to open $MANIFEST for reading: $!\n";
     open my $new, ">", $MANIFEST_NEW
         or die "Failed to open $MANIFEST_NEW for writing: $!\n";
+    my %keep = map +("cpan/$pkg_dir/$_" => 1), keys %new_files;
     my %gone = map +("cpan/$pkg_dir/$_" => 1), @gone;
     while (my $line = <$orig>) {
-        my ($file) = $line =~ /^(\S+)/
-            or die "Can't parse MANIFEST line: $line";
-        print $new $line if !$gone{$file};
+        chomp $line;
+        my ($file, $descr) = split /\t+/, $line;
```

Comment:
Would this be better as `split /\s+/, $line, 2`? While the current MANIFEST file always uses tabs, the format of the file allows any whitespace to divide the file names and the comments.

Reply:
>  the format of the file allows any whitespace to divide the file names and the comments.

No, that isn't correct for the core MANIFEST, it isn't the same format specification as ExtUtils::Manifest uses (assuming you are right and it allows it - I don't necessarily see why it would, on some platforms having spaces in file names in a manifest might actually be desirable). The core MANIFEST MUST be tabs and it cannot be spaces. We have tests to validate this. See https://github.com/Perl/perl5/blob/blead/t/porting/manifest.t#L58

Also for this case it doesn't matter if there are additional tabs in the file as the $descr is only used to check if there is /something/ there at all. FWIW, there is a single line in the current MANIFEST which has additional tabs after the separator tabs, which was obviously a case of duplication:

```
perl -lne'/(\S+)\t+([^\t]+)\t(.*)/ and print' MANIFEST
cpan/podlators/t/data/basic.cap			podlators test			podlators test
```

I have a patch to fix this. I am working on a patch to improve MANIFEST sorting and  formatting anyway, which will deal with this.

Reply:
ExtUtils::Manifest allows spaces, and allows using single quotes around a file name if it contains spaces.

In core, most of the programs that read MANIFEST accept spaces as a separator. A couple use ExtUtils::Manifest. A number also require there to be whitespace or a tab character to read the line properly.

Reply:
>In core, most of the programs that read MANIFEST accept spaces as a separator.

That may be. I would argue that they are just doing it wrong. *shrug*.

I don't plan to change this in this commit. If you want to address this discrepancy I think you should do so in another commit/PR which changes the test file and all the other tools that assume this file uses tabs.  As far as I can tell this is something where EU::Manifest was changed and core was not. IMO that is out of scope for this change.

Reply:
BTW, I don't see why this bothers you. If EU::Manifest and other tools assume that the file is "any whitespace" separated then that is compatible with the core being stricter and assuming it is tab separated. And if EU::Manifest has adopted a more complex file format with quotes around file names with spaces that just means it is harder to process and manage the manifest. With core using a simpler set of rules it is easier to deal with, and is still compatible with EU::Manifest. So nothing will break so long as someone doesn't assume you can use EU::Manifest to *write* the manifest file out, and we don't and shouldn't use EU::Manifest to write the manifest file, or sort it, or anything like that (or it would break our tests). So it seems to me that EU::Manifest is just enforcing a super set of the core format. There is no incompatibility there.

Reply:
The previous code split on white space, and most of the code dealing with the MANIFEST does the same, so it seemed reasonable behavior to maintain. And having two different MANIFEST formats is a bit weird.

I don't really care much though - my second comment was meant to be informational. And I did find some problems which will be worth addressing after this. Specifically, that a number of the scripts reading MANIFEST expect there to be whitespace or a tab on every entry, which currently isn't true. Not relevant to this PR though.

Reply:
> The previous code split on white space

FWIW, that isn't actually correct. The previous code didn't split at all, it just extracted the filename as a set of non-whitespace characters and ignored the description. So I'd say what it did is more compatible with the core-subset of the EU::Manifest format than it is with the EU::Manifest format itself. If it wanted to emulate that it should handle quoted filenames no?  Regardless I think you could argue that what we are doing now and before isn't /strictly/ speaking correct as \S matches \0. :-) 

The old logic was:
```
        my ($file) = $line =~ /^(\S+)/
            or die "Can't parse MANIFEST line: $line";
```